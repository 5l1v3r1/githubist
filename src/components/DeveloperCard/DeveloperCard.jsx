// @flow

import React from 'react';
import { Link } from 'react-router-dom';
import cx from 'classnames';

import { Avatar, Box } from '..';

import toDateString from '../../utils/toDateString';
import abbrNumber from '../../utils/abbrNumber';

import styles from './DeveloperCard.scss';

type Props = {
  rank?: number,
  name: string,
  username: string,
  profilePicture: string,
  company?: string,
  followers?: number,
  totalStarred?: number,
  repositoriesCount?: number,
  repoText?: string,
  location?: { name: string, slug: string },
  githubCreatedAt?: string,
};

const DeveloperCard = ({
  rank,
  name,
  username,
  profilePicture,
  company,
  followers,
  totalStarred,
  repositoriesCount,
  repoText,
  location,
  githubCreatedAt,
}: Props) => (
  <Box>
    <div className={styles.profile}>
      <Link to={`/${username}`} className={styles.avatar}>
        <Avatar size="small" url={profilePicture} alt={name} />
      </Link>

      <div className={styles.info}>
        <Link to={`/${username}`} className={styles.nameAndUsername}>
          {rank && <span className={styles.rank}>#{rank}</span>}
          <span className={styles.name}>{name}</span>
          <span className={styles.username}>{username}</span>
        </Link>
        {company && <p className={styles.company}>{company}</p>}
      </div>
    </div>

    <div className={styles.meta}>
      {location && (
        <Link to={`/location/${location.slug}`} className={cx(styles.metaItem, styles.location)}>
          {location.name}
        </Link>
      )}
      {typeof totalStarred !== 'undefined' && (
        <span className={styles.metaItem}>{abbrNumber(totalStarred)} Star&apos;lanma</span>
      )}
      {typeof followers !== 'undefined' && (
        <span className={cx(styles.metaItem)}>{abbrNumber(followers)} Takip√ßi</span>
      )}
      {typeof repositoriesCount !== 'undefined' && (
        <span className={cx(styles.metaItem)}>
          {repositoriesCount.toLocaleString()} {repoText}
        </span>
      )}
      {githubCreatedAt && (
        <span className={cx(styles.metaItem)}>{toDateString(githubCreatedAt)}</span>
      )}
    </div>
  </Box>
);

DeveloperCard.defaultProps = {
  rank: undefined,
  followers: undefined,
  totalStarred: undefined,
  location: undefined,
  githubCreatedAt: '',
  company: '',
  repositoriesCount: undefined,
  repoText: 'Repo',
};

export default DeveloperCard;
