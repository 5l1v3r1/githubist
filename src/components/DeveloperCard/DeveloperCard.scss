@import '../../theme/theme';

.profile {
  display: flex;
  flex-direction: row;
  align-items: center;
}

.avatar {
  line-height: 0;
}

.info {
  margin-left: map-get($spacers, normal);
}

.nameAndUsername {
  text-decoration: none;

  &:hover {
    color: $black-color;
    text-decoration: underline;
  }
}

.rank {
  margin-right: map-get($spacers, 2x-small);
}

.rank,
.name {
  color: $black-color;
  font-weight: 600;
}

.username {
  margin-left: map-get($spacers, 2x-small);
  color: $body-color;
}

.company {
  margin: 0;
  color: $body-color;
  font-size: map-get($font-sizes, small);
}

.meta {
  display: flex;
  flex-direction: row;
  margin: map-get($spacers, normal) 0 0 0;
}

.metaItem {
  margin-right: map-get($spacers, normal);
  font-size: map-get($font-sizes, small);
  font-weight: 600;
}

.location {
  color: $secondary-color;
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
}

.githubCreatedAt {
  margin-right: 0;
  margin-left: auto;
}
