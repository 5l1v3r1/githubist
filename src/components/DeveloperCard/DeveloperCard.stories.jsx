// @flow

import React from 'react';
import { storiesOf } from '@storybook/react';
import backgrounds from '@storybook/addon-backgrounds';
import { MemoryRouter } from 'react-router-dom';

import DeveloperCard from './DeveloperCard';

storiesOf('DeveloperCard', module)
  .addDecorator(backgrounds([{ name: 'light', value: '#fcfafa', default: true }]))
  .add('default', () => (
    <MemoryRouter>
      <div style={{ width: '500px', margin: '16px' }}>
        <DeveloperCard
          name="Alpcan Aydın"
          username="alpcanaydin"
          profilePicture="https://avatars2.githubusercontent.com/u/1801024?v=4"
          company="@Atolye15"
          followers={404}
          totalStarred={612}
          repositoriesCount={23}
          location={{ name: 'İzmir', slug: 'izmir' }}
        />
      </div>
    </MemoryRouter>
  ))
  .add('with rank', () => (
    <MemoryRouter>
      <div style={{ width: '500px', margin: '16px' }}>
        <DeveloperCard
          rank={1}
          name="Alpcan Aydın"
          username="alpcanaydin"
          profilePicture="https://avatars2.githubusercontent.com/u/1801024?v=4"
          company="@Atolye15"
          followers={404}
          totalStarred={612}
          repositoriesCount={23}
          location={{ name: 'İzmir', slug: 'izmir' }}
        />
      </div>
    </MemoryRouter>
  ))
  .add('with github creation date', () => (
    <MemoryRouter>
      <div style={{ width: '500px', margin: '16px' }}>
        <DeveloperCard
          name="Alpcan Aydın"
          username="alpcanaydin"
          profilePicture="https://avatars2.githubusercontent.com/u/1801024?v=4"
          company="@Atolye15"
          followers={404}
          totalStarred={612}
          repositoriesCount={23}
          location={{ name: 'İzmir', slug: 'izmir' }}
          githubCreatedAt="2012-05-31T22:02:23.000000Z"
        />
      </div>
    </MemoryRouter>
  ))
  .add('without company', () => (
    <MemoryRouter>
      <div style={{ width: '500px', margin: '16px' }}>
        <DeveloperCard
          name="Alpcan Aydın"
          username="alpcanaydin"
          profilePicture="https://avatars2.githubusercontent.com/u/1801024?v=4"
          totalStarred={612}
          followers={404}
          repositoriesCount={23}
          location={{ name: 'İzmir', slug: 'izmir' }}
        />
      </div>
    </MemoryRouter>
  ));
