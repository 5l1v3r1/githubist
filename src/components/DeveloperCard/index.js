// @flow

import DeveloperCard from './DeveloperCard';

export default DeveloperCard;
