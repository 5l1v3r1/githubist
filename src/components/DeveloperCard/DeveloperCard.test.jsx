// @flow

import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import DeveloperCard from './DeveloperCard';

it('renders correctly', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <DeveloperCard
        name="Alpcan Aydın"
        username="alpcanaydin"
        profilePicture="https://avatars2.githubusercontent.com/u/1801024?v=4"
        company="@Atolye15"
        totalStarred={612}
        followers={404}
        repositoriesCount={23}
        location={{ name: 'İzmir', slug: 'izmir' }}
      />
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly with rank', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <DeveloperCard
        rank={1}
        name="Alpcan Aydın"
        username="alpcanaydin"
        profilePicture="https://avatars2.githubusercontent.com/u/1801024?v=4"
        company="@Atolye15"
        totalStarred={612}
        followers={404}
        repositoriesCount={23}
        location={{ name: 'İzmir', slug: 'izmir' }}
      />
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly with github creation date', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <DeveloperCard
        name="Alpcan Aydın"
        username="alpcanaydin"
        profilePicture="https://avatars2.githubusercontent.com/u/1801024?v=4"
        company="@Atolye15"
        totalStarred={612}
        followers={404}
        repositoriesCount={23}
        location={{ name: 'İzmir', slug: 'izmir' }}
        githubCreatedAt="2012-05-31T22:02:23.000000Z"
      />
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly without company', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <DeveloperCard
        name="Alpcan Aydın"
        username="alpcanaydin"
        profilePicture="https://avatars2.githubusercontent.com/u/1801024?v=4"
        totalStarred={612}
        followers={404}
        repositoriesCount={23}
        location={{ name: 'İzmir', slug: 'izmir' }}
      />
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly with custom repo text', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <DeveloperCard
        name="Alpcan Aydın"
        username="alpcanaydin"
        profilePicture="https://avatars2.githubusercontent.com/u/1801024?v=4"
        totalStarred={612}
        followers={404}
        repositoriesCount={23}
        repoText="Java Reposu"
        location={{ name: 'İzmir', slug: 'izmir' }}
      />
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});
