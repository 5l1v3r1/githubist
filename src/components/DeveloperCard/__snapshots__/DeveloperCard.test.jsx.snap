// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <DeveloperCard
      company="@Atolye15"
      followers={404}
      githubCreatedAt=""
      location={
        Object {
          "name": "İzmir",
          "slug": "izmir",
        }
      }
      name="Alpcan Aydın"
      profilePicture="https://avatars2.githubusercontent.com/u/1801024?v=4"
      repoText="Repo"
      repositoriesCount={23}
      totalStarred={612}
      username="alpcanaydin"
    >
      <Box
        className=""
      >
        <div
          className=""
        >
          <div>
            <Link
              replace={false}
              to="/alpcanaydin"
            >
              <a
                href="/alpcanaydin"
                onClick={[Function]}
              >
                <Avatar
                  alt="Alpcan Aydın"
                  size="small"
                  url="https://avatars2.githubusercontent.com/u/1801024?v=4"
                >
                  <img
                    alt="Alpcan Aydın"
                    className="undefined"
                    src="https://avatars2.githubusercontent.com/u/1801024?v=4"
                  />
                </Avatar>
              </a>
            </Link>
            <div>
              <Link
                replace={false}
                to="/alpcanaydin"
              >
                <a
                  href="/alpcanaydin"
                  onClick={[Function]}
                >
                  <span>
                    Alpcan Aydın
                  </span>
                  <span>
                    alpcanaydin
                  </span>
                </a>
              </Link>
              <p>
                @Atolye15
              </p>
            </div>
          </div>
          <div>
            <Link
              className=""
              replace={false}
              to="/location/izmir"
            >
              <a
                className=""
                href="/location/izmir"
                onClick={[Function]}
              >
                İzmir
              </a>
            </Link>
            <span>
              612
               Star'lanma
            </span>
            <span>
              404
               Takipçi
            </span>
            <span>
              23
               
              Repo
            </span>
          </div>
        </div>
      </Box>
    </DeveloperCard>
  </Router>
</MemoryRouter>
`;

exports[`renders correctly with custom repo text 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <DeveloperCard
      company=""
      followers={404}
      githubCreatedAt=""
      location={
        Object {
          "name": "İzmir",
          "slug": "izmir",
        }
      }
      name="Alpcan Aydın"
      profilePicture="https://avatars2.githubusercontent.com/u/1801024?v=4"
      repoText="Java Reposu"
      repositoriesCount={23}
      totalStarred={612}
      username="alpcanaydin"
    >
      <Box
        className=""
      >
        <div
          className=""
        >
          <div>
            <Link
              replace={false}
              to="/alpcanaydin"
            >
              <a
                href="/alpcanaydin"
                onClick={[Function]}
              >
                <Avatar
                  alt="Alpcan Aydın"
                  size="small"
                  url="https://avatars2.githubusercontent.com/u/1801024?v=4"
                >
                  <img
                    alt="Alpcan Aydın"
                    className="undefined"
                    src="https://avatars2.githubusercontent.com/u/1801024?v=4"
                  />
                </Avatar>
              </a>
            </Link>
            <div>
              <Link
                replace={false}
                to="/alpcanaydin"
              >
                <a
                  href="/alpcanaydin"
                  onClick={[Function]}
                >
                  <span>
                    Alpcan Aydın
                  </span>
                  <span>
                    alpcanaydin
                  </span>
                </a>
              </Link>
            </div>
          </div>
          <div>
            <Link
              className=""
              replace={false}
              to="/location/izmir"
            >
              <a
                className=""
                href="/location/izmir"
                onClick={[Function]}
              >
                İzmir
              </a>
            </Link>
            <span>
              612
               Star'lanma
            </span>
            <span>
              404
               Takipçi
            </span>
            <span>
              23
               
              Java Reposu
            </span>
          </div>
        </div>
      </Box>
    </DeveloperCard>
  </Router>
</MemoryRouter>
`;

exports[`renders correctly with github creation date 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <DeveloperCard
      company="@Atolye15"
      followers={404}
      githubCreatedAt="2012-05-31T22:02:23.000000Z"
      location={
        Object {
          "name": "İzmir",
          "slug": "izmir",
        }
      }
      name="Alpcan Aydın"
      profilePicture="https://avatars2.githubusercontent.com/u/1801024?v=4"
      repoText="Repo"
      repositoriesCount={23}
      totalStarred={612}
      username="alpcanaydin"
    >
      <Box
        className=""
      >
        <div
          className=""
        >
          <div>
            <Link
              replace={false}
              to="/alpcanaydin"
            >
              <a
                href="/alpcanaydin"
                onClick={[Function]}
              >
                <Avatar
                  alt="Alpcan Aydın"
                  size="small"
                  url="https://avatars2.githubusercontent.com/u/1801024?v=4"
                >
                  <img
                    alt="Alpcan Aydın"
                    className="undefined"
                    src="https://avatars2.githubusercontent.com/u/1801024?v=4"
                  />
                </Avatar>
              </a>
            </Link>
            <div>
              <Link
                replace={false}
                to="/alpcanaydin"
              >
                <a
                  href="/alpcanaydin"
                  onClick={[Function]}
                >
                  <span>
                    Alpcan Aydın
                  </span>
                  <span>
                    alpcanaydin
                  </span>
                </a>
              </Link>
              <p>
                @Atolye15
              </p>
            </div>
          </div>
          <div>
            <Link
              className=""
              replace={false}
              to="/location/izmir"
            >
              <a
                className=""
                href="/location/izmir"
                onClick={[Function]}
              >
                İzmir
              </a>
            </Link>
            <span>
              612
               Star'lanma
            </span>
            <span>
              404
               Takipçi
            </span>
            <span>
              23
               
              Repo
            </span>
            <span
              className=""
            >
              1 Haziran 2012
            </span>
          </div>
        </div>
      </Box>
    </DeveloperCard>
  </Router>
</MemoryRouter>
`;

exports[`renders correctly with rank 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <DeveloperCard
      company="@Atolye15"
      followers={404}
      githubCreatedAt=""
      location={
        Object {
          "name": "İzmir",
          "slug": "izmir",
        }
      }
      name="Alpcan Aydın"
      profilePicture="https://avatars2.githubusercontent.com/u/1801024?v=4"
      rank={1}
      repoText="Repo"
      repositoriesCount={23}
      totalStarred={612}
      username="alpcanaydin"
    >
      <Box
        className=""
      >
        <div
          className=""
        >
          <div>
            <Link
              replace={false}
              to="/alpcanaydin"
            >
              <a
                href="/alpcanaydin"
                onClick={[Function]}
              >
                <Avatar
                  alt="Alpcan Aydın"
                  size="small"
                  url="https://avatars2.githubusercontent.com/u/1801024?v=4"
                >
                  <img
                    alt="Alpcan Aydın"
                    className="undefined"
                    src="https://avatars2.githubusercontent.com/u/1801024?v=4"
                  />
                </Avatar>
              </a>
            </Link>
            <div>
              <Link
                replace={false}
                to="/alpcanaydin"
              >
                <a
                  href="/alpcanaydin"
                  onClick={[Function]}
                >
                  <span>
                    #
                    1
                  </span>
                  <span>
                    Alpcan Aydın
                  </span>
                  <span>
                    alpcanaydin
                  </span>
                </a>
              </Link>
              <p>
                @Atolye15
              </p>
            </div>
          </div>
          <div>
            <Link
              className=""
              replace={false}
              to="/location/izmir"
            >
              <a
                className=""
                href="/location/izmir"
                onClick={[Function]}
              >
                İzmir
              </a>
            </Link>
            <span>
              612
               Star'lanma
            </span>
            <span>
              404
               Takipçi
            </span>
            <span>
              23
               
              Repo
            </span>
          </div>
        </div>
      </Box>
    </DeveloperCard>
  </Router>
</MemoryRouter>
`;

exports[`renders correctly without company 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <DeveloperCard
      company=""
      followers={404}
      githubCreatedAt=""
      location={
        Object {
          "name": "İzmir",
          "slug": "izmir",
        }
      }
      name="Alpcan Aydın"
      profilePicture="https://avatars2.githubusercontent.com/u/1801024?v=4"
      repoText="Repo"
      repositoriesCount={23}
      totalStarred={612}
      username="alpcanaydin"
    >
      <Box
        className=""
      >
        <div
          className=""
        >
          <div>
            <Link
              replace={false}
              to="/alpcanaydin"
            >
              <a
                href="/alpcanaydin"
                onClick={[Function]}
              >
                <Avatar
                  alt="Alpcan Aydın"
                  size="small"
                  url="https://avatars2.githubusercontent.com/u/1801024?v=4"
                >
                  <img
                    alt="Alpcan Aydın"
                    className="undefined"
                    src="https://avatars2.githubusercontent.com/u/1801024?v=4"
                  />
                </Avatar>
              </a>
            </Link>
            <div>
              <Link
                replace={false}
                to="/alpcanaydin"
              >
                <a
                  href="/alpcanaydin"
                  onClick={[Function]}
                >
                  <span>
                    Alpcan Aydın
                  </span>
                  <span>
                    alpcanaydin
                  </span>
                </a>
              </Link>
            </div>
          </div>
          <div>
            <Link
              className=""
              replace={false}
              to="/location/izmir"
            >
              <a
                className=""
                href="/location/izmir"
                onClick={[Function]}
              >
                İzmir
              </a>
            </Link>
            <span>
              612
               Star'lanma
            </span>
            <span>
              404
               Takipçi
            </span>
            <span>
              23
               
              Repo
            </span>
          </div>
        </div>
      </Box>
    </DeveloperCard>
  </Router>
</MemoryRouter>
`;
